nx-X11/programs/Xserver/hw/nxagent/Events.c

```diff
@@ -2761,7 +2761,7 @@ int nxagentHandleClientMessageEvent(XEvent *X, enum HandleEventResult *result)
       else
       {
         #ifdef TEST
-        fprintf(stderr, "Events: WM_DELETE_WINDOW arrived Atom = %ld.\n", wmAtom);
```

Comment:
Seems to be okay as well.

Events implementation is scary and complicated - all Events are 32 bytes wide, but how the data is interpreted depends on the events type.

The crucial part is this one:
```C
 wmAtom = (Atom) X -> xclient.data.l[0];
```

Since `X->xclient.data.l[0]` is typed `long`, it will be accessed as such and we don't need to care about it. `Atom` will be a `CARD32` object, which means that a conversion needs to take place, but since `XID` in X server contexts are always `CARD32` objects and clients are confined to using a range predefined by the server only, the conversion should work fine.

`wmAtom` isn't actually used later on, which is also good.